src/lib/sql/sql_identifier_resolver.cpp

```diff
@@ -17,8 +17,10 @@ void SQLIdentifierResolver::add_column_name(const std::shared_ptr<AbstractExpres
 }
 
 void SQLIdentifierResolver::reset_column_names(const std::shared_ptr<opossum::AbstractExpression>& expression) {
-  auto& entry = _find_expression_entry(expression);
-  entry.column_names.clear();
+  auto entry_iter = std::find_if(_entries.begin(), _entries.end(),
+                                 [&](const auto& entry) { return *entry.expression == *expression; });
+  if (entry_iter == _entries.end()) return;
+  entry_iter->column_names.clear();
```

Comment:
Taken from `_find_expression_entry`, which was not used anywhere else. Only change is that the assertion is removed. An expression might not have a name yet (because it is, e.g., an aggregate), but we want to call `reset_column_names` indiscriminately.